tests/base/src/test/java/org/wildfly/security/ssl/SSLAuthenticationTest.java

```diff
@@ -731,60 +725,93 @@ public void testClientSideOcsp() throws Exception {
                 .setClientMode(true)
                 .build().create();
 
-        performConnectionTest(serverContextGood, clientContext, true);
-        performConnectionTest(serverContextRevoked, clientContext, false);
+
+        testCommunication(serverContextGood, clientContext, "OU=Elytron,O=Elytron,C=UK,ST=Elytron,CN=ocspCheckedGood",
+                null, null);
+        try {
+            testCommunication(serverContextRevoked, clientContext, null, null, null);
+            Assert.fail("Expected SSLHandshakeException not thrown");
+        } catch (SSLHandshakeException expected) {
+            //expected
+        }
     }
 
-    private SecurityIdentity performConnectionTest(SSLContext serverContext, String clientUri, boolean expectValid) throws Exception {
+    private void performConnectionTest(SSLContext serverContext, String clientUri, boolean expectValid, String expectedServerPrincipal, String expectedClientPrincipal, String expectedCipherSuite) throws Throwable {
         System.setProperty("wildfly.config.url", SSLAuthenticationTest.class.getResource("wildfly-ssl-test-config-v1_1.xml").toExternalForm());
         AccessController.doPrivileged((PrivilegedAction<Integer>) () -> Security.insertProviderAt(WildFlyElytronPasswordProvider.getInstance(), 1));
 
         AuthenticationContext context = AuthenticationContext.getContextManager().get();
         AuthenticationContextConfigurationClient contextConfigurationClient = AccessController.doPrivileged(AuthenticationContextConfigurationClient.ACTION);
         SSLContext clientContext = contextConfigurationClient.getSSLContext(URI.create(clientUri), context);
 
-        return performConnectionTest(serverContext, clientContext, expectValid);
+        try {
+            testCommunication(serverContext, clientContext, expectedServerPrincipal, expectedClientPrincipal, expectedCipherSuite);
+            if(!expectValid) Assert.fail("Expected SSLHandshakeException not thrown");
+        } catch (SSLHandshakeException expected) {
+            //expected
```

Comment:
I think we need to be a bit careful here because if ```expectValid``` is true but ```testCommunication``` happens to throw an ```SSLHandshakeException```, it will erroneously be caught here. One thing you could do is just call ```testCommunication``` here and then wrap the calls to ```performConnectionTest``` with a ```try/catch``` in the tests that are asserting that an ```SSLHandshakeException``` should actually be thrown.

Reply:
Fixed, thanks!